modules/tagorasBidAdapter.js

```diff
@@ -0,0 +1,342 @@
+        break;
+      default:
+        payloadRef[key] = userId;
+    }
+  });
+}
+
+function buildRequests(validBidRequests, bidderRequest) {
+  const topWindowUrl = bidderRequest.refererInfo.page || bidderRequest.refererInfo.topmostLocation;
+  const bidderTimeout = config.getConfig('bidderTimeout');
+  const requests = [];
+  validBidRequests.forEach(validBidRequest => {
+    const sizes = parseSizesInput(validBidRequest.sizes);
+    const request = buildRequest(validBidRequest, topWindowUrl, sizes, bidderRequest, bidderTimeout);
+    requests.push(request);
+  });
+  return requests;
+}
+
+function interpretResponse(serverResponse, request) {
+  if (!serverResponse || !serverResponse.body) {
+    return [];
+  }
+  const {bidId} = request.data;
+  const {results} = serverResponse.body;
+
+  let output = [];
+
+  try {
+    results.forEach(result => {
+      const {
+        creativeId,
+        ad,
+        price,
+        exp,
+        width,
+        height,
+        currency,
+        metaData,
+        advertiserDomains,
+        mediaType = BANNER
+      } = result;
+      if (!ad || !price) {
+        return;
+      }
+
+      const response = {
+        requestId: bidId,
+        cpm: price,
+        width: width,
+        height: height,
+        creativeId: creativeId,
+        currency: currency || CURRENCY,
+        netRevenue: true,
+        ttl: exp || TTL_SECONDS,
+      };
+
+      if (metaData) {
+        Object.assign(response, {
+          meta: metaData
+        })
+      } else {
+        Object.assign(response, {
+          meta: {
+            advertiserDomains: advertiserDomains || []
+          }
+        })
+      }
+
+      if (mediaType === BANNER) {
+        Object.assign(response, {
+          ad: ad,
+        });
+      } else {
+        Object.assign(response, {
+          vastXml: ad,
+          mediaType: VIDEO
+        });
+      }
+      output.push(response);
+    });
+    return output;
+  } catch (e) {
+    return [];
+  }
+}
+
+function getUserSyncs(syncOptions, responses, gdprConsent = {}, uspConsent = '', gppConsent = {}) {
+  let syncs = [];
+  const {iframeEnabled, pixelEnabled} = syncOptions;
+  const {gdprApplies, consentString = ''} = gdprConsent;
+  const {gppString, applicableSections} = gppConsent;
+
+  const cidArr = responses.filter(resp => deepAccess(resp, 'body.cid')).map(resp => resp.body.cid).filter(uniques);
+  let params = `?cid=${encodeURIComponent(cidArr.join(','))}&gdpr=${gdprApplies ? 1 : 0}&gdpr_consent=${encodeURIComponent(consentString || '')}&us_privacy=${encodeURIComponent(uspConsent || '')}`
+
+  if (gppString && applicableSections?.length) {
+    params += '&gpp=' + encodeURIComponent(gppString);
+    params += '&gpp_sid=' + encodeURIComponent(applicableSections.join(','));
+  }
+
+  if (iframeEnabled) {
+    syncs.push({
+      type: 'iframe',
+      url: `https://sync.tagoras.io/api/sync/iframe/${params}`
+    });
+  }
+  if (pixelEnabled) {
```

Comment:
Can we make this an either or and prioritize one over the other to avoid adding both syncs when all are enabled